From 28b46920144e368adc1d39512289142312eb9026 Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Tue, 2 Feb 2021 19:59:48 +0100
Subject: kern/lockdown: Set a variable if the GRUB is locked down

It may be useful for scripts to determine whether the GRUB is locked
down or not. Add the lockdown variable which is set to "y" when the GRUB
is locked down.

Suggested-by: Dimitri John Ledkov <xnox@ubuntu.com>
Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>

Patch-Name: 2021-02-security/002-kern-lockdown-Set-a-variable-if-the-GRUB-is-locked-down.patch
---
 docs/grub.texi            | 3 +++
 grub-core/kern/lockdown.c | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/docs/grub.texi b/docs/grub.texi
index fe2fddb63..c2ef67756 100644
--- a/docs/grub.texi
+++ b/docs/grub.texi
@@ -5666,6 +5666,9 @@ The GRUB can be locked down when booted on a secure boot environment, for exampl
 if the UEFI secure boot is enabled. On a locked down configuration, the GRUB will
 be restricted and some operations/commands cannot be executed.
 
+The @samp{lockdown} variable is set to @samp{y} when the GRUB is locked down.
+Otherwise it does not exit.
+
 @node Platform limitations
 @chapter Platform limitations
 
diff --git a/grub-core/kern/lockdown.c b/grub-core/kern/lockdown.c
index 7e7d99ae8..2d1815c34 100644
--- a/grub-core/kern/lockdown.c
+++ b/grub-core/kern/lockdown.c
@@ -18,6 +18,7 @@
  */
 
 #include <grub/dl.h>
+#include <grub/env.h>
 #include <grub/file.h>
 #include <grub/lockdown.h>
 
@@ -84,6 +85,9 @@ grub_lockdown (void)
 #if 0
   grub_verifier_register (&lockdown_verifier);
 #endif
+
+  grub_env_set ("lockdown", "y");
+  grub_env_export ("lockdown");
 }
 
 int
